patches/minecraft/net/minecraft/item/Item.java.patch

```diff
@@ -354,12 +354,14 @@
 +
 +    /**
 +     * Determines if the specific ItemStack can be placed in the specified armor slot.
++     * @deprecated Use {@code armorType == EntityLiving.getSlotForItemStack(stack)} instead.
```

Comment:
You should probably link to `Item::getEquipmentSlot` here instead, since most uses of this method are overriding it, not calling it.